app/src/main/java/org/eyeseetea/malariacare/domain/usecase/AGetReviewValuesBySurveyIdUseCaseStrategy.java

```diff
@@ -0,0 +1,81 @@
+package org.eyeseetea.malariacare.domain.usecase;
+
+
+import com.google.common.collect.Iterables;
+
+import org.eyeseetea.malariacare.domain.boundary.executors.IAsyncExecutor;
+import org.eyeseetea.malariacare.domain.boundary.executors.IMainExecutor;
+import org.eyeseetea.malariacare.domain.boundary.repositories.IValueRepository;
+import org.eyeseetea.malariacare.domain.entity.Value;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public abstract class AGetReviewValuesBySurveyIdUseCaseStrategy implements UseCase {
+    private IMainExecutor mMainExecutor;
+    private IAsyncExecutor mAsyncExecutor;
+    private Callback mCallback;
+    private IValueRepository mIValueRepository;
+    private long mSurveyId;
+
+
+    public AGetReviewValuesBySurveyIdUseCaseStrategy(
+            IMainExecutor mainExecutor,
+            IAsyncExecutor asyncExecutor,
+            IValueRepository IValueRepository) {
+        mMainExecutor = mainExecutor;
+        mAsyncExecutor = asyncExecutor;
+        mIValueRepository = IValueRepository;
+    }
+
+    public void execute(Callback callback, long surveyId) {
+        mCallback = callback;
+        mSurveyId = surveyId;
+        mAsyncExecutor.run(this);
+    }
+
+    @Override
+    public void run() {
+        List<Value> values = mIValueRepository.getValuesFromSurvey(mSurveyId);
+        Iterator<String> colorIterator = Iterables.cycle(
```

Comment:
The logic to create a unique color list that appears on review screen is not a domain logic or presentation logic, it´s a painting logic or view logic then we should move it to activity, fragment ...